### DISCUSSION TITLE
Pass props from TOP to BOTTOM

### DISCUSSION THREAD

**Discussion by Merynek** [vercel/next.js#61192] [App Router]
Hello,
I want share any data for each request, so I create object with these data in rootLayout
```
export interface IProps {
    children: React.ReactNode;
    data: any;
}

export default function RootLayout(props: IProps) {
    const {children} = props;
    props.data = {item: "hello"} // I know this is bad (only for test)
    return children;
}
```
Now I need access to data page or in layout on server side.
```
export default function Layout(props: IProps) {
    const {children, data} = props;
    console.log("Data", data);
    return children;
}
```
Is there any way for share data per request or can I extend props for all children of root layout (on server only)?

**Comment by Nadeera3784** [comment]
Hi @Merynek ,
How about using Context API?

**Reply by Merynek** [reply]
Context Api is not available on server
